lib/matplotlib/style/core.py

```diff
@@ -66,22 +67,92 @@ def use(style):
     else:
         styles = style
 
+    styles = insert_parent_styles(styles)
+
     for style in styles:
         if not cbook.is_string_like(style):
             mpl.rcParams.update(style)
             continue
 
-        if style in library:
-            mpl.rcParams.update(library[style])
+        mpl.rcParams.update(get_style_parameters(style))
+
+
+def insert_parent_styles(styles):
+    """Returns a list with all the styles and parents styles called"""
+    tmp_styles = []
+    for style in styles:
+        style = _get_parent_styles_from_str_or_dict(style)
+        tmp_styles.extend(style)
+    return tmp_styles[:]
```

Comment:
Well, the result is the same, except in this case, we just need to add a list at the end of an other one, hence the use of `extend`. I wrote the `_list_flatten` function to be able to flatten the list after inserting objects in the middle of it, because having strings was giving me some trouble. It is not necessary to use it here, but I can change it if you think it would be clearer.